h2o-automl/src/main/java/ai/h2o/automl/targetencoding/FrameToTETableTask.java

```diff
@@ -0,0 +1,36 @@
+package ai.h2o.automl.targetencoding;
+
+import water.MRTask;
+import water.fvec.Chunk;
+import water.util.IcedHashMap;
+
+public class FrameToTETableTask extends MRTask<FrameToTETableTask> {
+  
+  // IcedHashMap does not support integer keys so we will store indices as strings.
+  public IcedHashMap<String, TEComponents> _table = new IcedHashMap<>();
```

Comment:
Probably I can try to rewrite this with arrays as it is done in my PR for Broadcast Join https://github.com/h2oai/h2o-3/pull/3233